### PR TITLE
Drop redundant -Xsource checks

### PR DESCRIPTION
**Pull request by ashawley**
While studying the changes for -Xfuture, in #7328, it seemed like code paths for 2.12 and 2.13 should be weeded out.  Is there a point to keeping them around?  I presume these are unused code paths that can be removed?  Is there a requirement to preserve certain 2.11 and 2.12 behaviors in 2.13?

While we're at it, would it be beneficial to add `isScala215` before 2.13 is cut (and the `2.14.x` branch begins)?  I know fixes often get backported and forward ported between branches, so it seems like this would be helpful to get cleaned up.  If this change likely makes that work more difficult, than I'll withdraw.

WIP: I've only run the partest tests, so something could come up in the test suite.

### CODE REVIEW

File: test/files/neg/badtok-1-212.scala [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// scalac: -Xsource:2.12 -deprecation -Xfatal-warnings
+// scalac: -deprecation -Xfatal-warnings
```
Comment (som-snytt) [inline test/files/neg/badtok-1-212.scala]: This test is for 2.12 messaging; the other similar test is for 2.13.

Comment (ashawley) [reply test/files/neg/badtok-1-212.scala]: Ok, I've deleted this redundant test.

**Comment by NthPortal** [comment]
it's my understanding that there is not supposed to be a 2.15

**Comment by som-snytt** [comment]
I don't know if English has an adequate verbal tense or temporal signifier, but it will have been the case that there wasn't supposed to have been a 2.15.

I just noticed that 2.14 is Valentine's Day. I hope they make 2.14 Valentine's-themed, with lots of ❤️ emojis and 😍 and 😻 and a 💟 label for issues. @SethTisue can you make that happen? or maybe just having that as a long-term goal will help you get through 2.13.

**Comment by dwijnand** [comment]
While I encourage the forward-thinking that brought to preempting 2.15 considerations in the 2.13 branch, I think in this case it would cause more confusion.  The reason is that with trying to close the distance between 2.14 and 3.0 having a "2.15" variable would probably make it more complicated to reason about.  We can always add it later and back or forward port it, with the test suite as guidance to where behaviours must be tweaked.

**Comment by ashawley** [comment]
I wasn't aware that 2.15 isn't planned.  I had heard it might be necessary as a stepping stone to progressively migrate libraries and code bases to 3.0.  I won't bother adding -Xsource:2.15, then.

**Comment by lrytz** [comment]
Let's also do this after 2.13.0 is out.

**Comment by ashawley** [comment]
It's permissible to break the compiler and reflection APIs, albeit in smalls ways, in a 2.13 minor release?

**Comment by lrytz** [comment]
Yes, in principle that's allowed

**Comment by dwijnand** [comment]
Not public scala.reflect APIs, AFAIK.

**Comment by lrytz** [comment]
Right; in theory we could break it because scala.reflect is experimental, but we usually don't.

**Comment by dwijnand** [comment]
TIL https://github.com/scala/scala/blob/ff05480394b00007d66381d20e583ad0e3ac8c2a/src/reflect/scala/reflect/api/package.scala#L19

**Comment by ashawley** [comment]
Wondering if @retronym thinks this is wise or not.

**Comment by retronym** [comment]
I think we should leave these in until 2.14.x as a potential migration tool for people coming from 2.12 to 2.13. I realise that we haven't captured all compiler changes under this flag so one could argue that this feature is only half-baked and we should rethink it, but let's err on the side of caution for now.

**Comment by retronym** [comment]
(Sorry about the slow reply on this one)

**Pull request closed without merging**